### PR TITLE
docker: create docker image

### PR DESCRIPTION
**Pull request by giose86**
In this PR i have:

- added a dockerfile.
- workdir refactoring: Made gremlins able to work within docker container.
- goreleaser: added the configuration to build the gremlins docker image and make it available for download on docker hub.

### CODE REVIEW

**Comment by k3rn31** [comment]
We should add Docker usage instructions in documentation. For example here: https://github.com/go-gremlins/gremlins/blob/main/docs/docs/install.md

I'd also add some specific CI usage instructions, but for that we can open a separate issue/pr.

**Review by k3rn31 — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -10,13 +10,23 @@ permissions:
...
           fetch-depth: 0
       - name: Fetch all tags
         run: git fetch --force --tags
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Docker Login
+        uses: docker/login-action@v1
+        with:
+          registry: docker.io
+          username: ${{ github.repository_owner }}
```
[the remark this answers is not in the document]
Comment (giose86) [reply .github/workflows/release.yml:28]: ok, i'll fix that

File: .goreleaser.yaml [hunk 2]
```diff
@@ -12,16 +12,50 @@ builds:
...
   draft: true
   prerelease: auto
+dockers:
+  - image_templates: ["docker.io/gogremlins/{{ .ProjectName }}:{{ .Version }}-386"]
+    use: buildx
+    build_flag_templates:
+      - --platform=linux/386
+      - --label=org.opencontainers.image.title={{ .ProjectName }}
+      - --label=org.opencontainers.image.description={{ .ProjectName }}
+      - --label=org.opencontainers.image.url=https://github.com/gogremlins/{{ .ProjectName }}
```
[the remark this answers is not in the document]
Comment (giose86) [reply .goreleaser.yaml:32]: ok, i'll fix that :)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FROM golang:1.18
+
+COPY gremlins /usr/bin
+RUN adduser --disabled-password --gecos "" nonroot
```
[the remark this answers is not in the document]
Comment (giose86) [reply Dockerfile:4]: it is added by default by adduser command

[the remark this answers is not in the document]
Comment (k3rn31) [reply Dockerfile:4]: Good, so we it should be
```
USER nonroot:nonroot
```

**Pull request closed without merging**